cmd/simulator/txs/agent.go

```diff
@@ -75,11 +92,71 @@ func (a issueNAgent[T]) Execute(ctx context.Context) error {
 			}
 			txs = append(txs, tx)
 		}
+		issuedEnd := time.Now()
+		issuedDuration := issuedEnd.Sub(issuedStart)
+		log.Info("Issuance Batch Done", "batch", batchI, "time", issuedDuration.Seconds())
+		// Add the issuance batch time to the total issuedTime
+		issuedTime = issuedTime.Add(issuedDuration)
 
+		// Start confirmation batch
+		confirmedStart := time.Now()
 		for i, tx := range txs {
 			if err := a.worker.ConfirmTx(ctx, tx); err != nil {
 				return fmt.Errorf("failed to await transaction %d: %w", i, err)
 			}
+			confirmedCount++
+			if confirmedCount == totalTxs {
```

Comment:
I think we should switch the chan end from `return a.worker.Close(ctx)` to breaking the loop and do this after, or less preferable switching this to a defer statement

Reply:
Discussed offline